jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/ResultAttribute.java

```diff
@@ -22,61 +22,50 @@
  */
 package com.googlecode.jmxtrans.model;
 
+import lombok.Getter;
 import lombok.SneakyThrows;
 import lombok.ToString;
 import org.apache.commons.lang.StringUtils;
 
 import javax.annotation.Nonnull;
-import java.lang.reflect.Method;
 import java.util.Map;
 
-import static org.apache.commons.lang.StringUtils.capitalize;
-
 /**
- * Enumerates the attributes of {@link Result}
- * 
+ * Attributes of {@link Result}
+ *
  * @author Simon Hutchinson
  *         <a href="https://github.com/sihutch">github.com/sihutch</a>
  *
  */
 @ToString
-public enum ResultAttribute {
-
-	TYPE_NAME("typeName"), OBJ_DOMAIN("objDomain"), CLASS_NAME("className"), ATTRIBUTE_NAME("attributeName");
+public abstract class ResultAttribute {
 
+	/**
+	 * Attribute name in camel case
+	 */
+	@Getter
 	@Nonnull
 	private String attributeName;
```

Comment:
Not really related to your change, but it looks that attributeName could be `final`.